File: src/main/java/com/example/helloworld/HelloWorldController.java
package com.example.helloworld;

import org.slf4j.Logger;
import org.slf4j.LoggerFactory;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.PostMapping;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RestController;

@RestController
public class HelloWorldController {
    Logger logger = LoggerFactory.getLogger(HelloWorldController.class);

    @GetMapping("/hello")
    public String hello() {
        return "Hello, World!";
    }

    @PostMapping("/hello")
    public String helloPost(@RequestBody String payload) {
        logger.info("Received payload: " + payload);
        return "You just called the POST method at '/hello'!\n";
    }
}

File: src/main/kotlin/com/example/helloworld/HelloWorldController.kt
package com.example.helloworld

import org.slf4j.Logger
import org.slf4j.LoggerFactory
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.PostMapping
import org.springframework.web.bind.annotation.RequestBody
import org.springframework.web.bind.annotation.RestController

@RestController
class HelloWorldController {

    private val logger: Logger = LoggerFactory.getLogger(HelloWorldController::class.java)

    @GetMapping("/hello")
    fun hello(): String {
        return "Hello, World!"
    }

    @PostMapping("/hello")
    fun helloPost(@RequestBody payload: String): String {
        logger.info("Received payload: $payload")
        return "You just called the POST method at '/hello'!\n"
    }
}